src/components/AddComment.jsx

```diff
@@ -0,0 +1,49 @@
+import { useState } from "react";
+import { addCommentHandler, getComments } from "../utils/api-requests";
+
+export function AddComment({
+  toggleAddCommentModal,
+  article_id,
+  username,
+  setComments,
+}) {
+  const [newComment, setNewComment] = useState("");
+  const [isLoading, setIsLoading] = useState(false);
+
+  async function newCommentHandler(e) {
+    e.preventDefault();
+    setIsLoading(true);
+
+    try {
+      await addCommentHandler(article_id, newComment, username);
+      const updatedComments = await getComments(article_id);
```

Comment:
Rather than re-fetching the comments here, we can manually add the comment to the comments array using the setter as optimistic rendering. If the request fails, we remove the comment from the array to undo the optimistic rendering and therefore we don't need to make the additional api request :)